lianad/src/commands/mod.rs

```
@@ -359,7 +370,126 @@ impl DaemonControl {
+
+        let persister = SenderPersister::new(Arc::new(self.db.clone())).unwrap();
+        let _sender = SenderBuilder::new(psbt.clone(), uri)
+            .build_recommended(FeeRate::BROADCAST_MIN)
+            .save(&persister)
+            .unwrap();
+
+        persister.update_metadata(
+            Some(PayjoinStatus::Pending),
+            Some(psbt.unsigned_tx.compute_txid()),
+            Some(psbt.clone()),
+            Some(bip21.clone()),
+        );
+
+        Ok(())
+    }
+
+    /// Get Payjoin URI (BIP21) and its sender/receiver status by txid
+    pub fn get_payjoin_info(&self, txid: &Txid) -> Result<Option<PayjoinInfo>, CommandError> {
+        let mut db_conn = self.db.connection();
+
+        let mut receiver_status = None;
+        for (_, session) in db_conn.payjoin_get_all_receiver_sessions() {
+            let SessionMetadata {
+                status, maybe_txid, ..
+            } = session.metadata.clone();
+            if let Some(db_txid) = maybe_txid {
+                if &db_txid == txid {
+                    receiver_status = Some(status);
+                    break;
+                }
+            }
+        }
+
+        let mut bip21 = String::new();
+        let mut sender_status = None;
+        for (_, session) in db_conn.payjoin_get_all_sender_sessions() {
+            let SessionMetadata {
+                status,
+                maybe_txid,
+                maybe_bip21,
+                ..
+            } = session.metadata.clone();
+            if let Some(db_txid) = maybe_txid {
+                if &db_txid == txid {
+                    sender_status = Some(status);
+                    bip21 = maybe_bip21.unwrap_or_default();
+                    break;
+                }
+            }
+        }
+
+        if receiver_status.is_some() || sender_status.is_some() {
+            Ok(Some(PayjoinInfo {
+                bip21,
+                sender_status,
+                receiver_status,
```

I dont understand how there can be both a sender a receiver status for the same payjoin. I think it would be fine to simplify this to one status for both. This should be attained by replaying the log for said session, if the session state is not invalid then its pending.If it is it has failed and we can return the corresponding error message as well.